### PR TITLE
add location

### PR DESCRIPTION
**Pull request by dmitrytulba**
# Changes

> Describe your changes in a short and concise list.

# Related issue

> Reference related issues, and use one of the [closing keywords, e.g. closes or fixes](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to link the corresponding issue, if any.

# Additional information 

> Provide additional information for this change, if needed.

# Pre-Merge checklist

- [ ] New code is tested
- [ ] Present and new tests pass
- [ ] Documentation is updated
- [ ] The build is working without errors
- [ ] No new Sonar issues introduced
- [ ] Changelog is updated
- [ ] Code has been reviewed

### CODE REVIEW

File: rest-openehr/src/main/java/org/ehrbase/rest/openehr/OpenehrDefinitionQueryController.java [hunk 1]
```diff
@@ -181,7 +181,7 @@ private ResponseEntity<QueryDefinitionResponseData> getPutDefenitionResponseEnti
         } else if (TEXT_PLAIN.isCompatibleWith(mediaType)) {
             HttpHeaders respHeaders = new HttpHeaders();
             respHeaders.setContentType(APPLICATION_JSON);
-            respHeaders.setLocation(getRequestUri());
+            respHeaders.setLocation(getLocationUri(DEFINITION, QUERY, storedQuery.getVersion()));
```
Comment (dmitrytulba) [inline rest-openehr/src/main/java/org/ehrbase/rest/openehr/OpenehrDefinitionQueryController.java:184]: @dmitrytulba check it

File: rest-openehr/src/main/java/org/ehrbase/rest/BaseController.java [hunk 2]
```diff
@@ -113,15 +117,38 @@ public Map<String, Map<String, String>> add2MetaMap(
         return metaMap;
     }
 
+    /**
+     * This method is intended to return the base URI, however it is currently returning the request URI.
+     *
+     * @deprecated This method is deprecated and should not be used in favor of a more updated version.
+     */
+    @Deprecated(forRemoval = true)
```
Comment (HolgerReiseVSys) [inline rest-openehr/src/main/java/org/ehrbase/rest/BaseController.java:125]: Is there a reason why we cannot just remove it?

File: rest-openehr/src/main/java/org/ehrbase/rest/BaseController.java [hunk 3]
```diff
@@ -113,15 +117,38 @@ public Map<String, Map<String, String>> add2MetaMap(
...
+        return ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString();
     }
 
+    @Value(API_CONTEXT_PATH_WITH_VERSION)
+    protected String apiContextPathWithVersion;
+
+    /**
+     * Returns a URI for list of segments.
+     * The segments are appended to the base path and encoded to ensure safe usage in a URI.
+     * This method uses the UriComponentsBuilder from Spring Framework to construct the URI.
```
Comment (HolgerReiseVSys) [inline rest-openehr/src/main/java/org/ehrbase/rest/BaseController.java:140]: I suppose mentioning UriComponentsBuilder is not needed

File: rest-openehr/src/main/java/org/ehrbase/rest/BaseController.java [hunk 4]
```diff
@@ -113,15 +117,38 @@ public Map<String, Map<String, String>> add2MetaMap(
...
+     *
+     * @param pathSegments List of segments to append to the base URL
+     * @return URI for the given base URL and segments
+     */
+    protected URI getLocationUri(String... pathSegments) {
```
Comment (HolgerReiseVSys) [inline rest-openehr/src/main/java/org/ehrbase/rest/BaseController.java:145]: Maybe rename to `createLocationUri`?

**Review by HolgerReiseVSys — CHANGES_REQUESTED** [review]
Looks good overall, just a bit minor cleanup

**Comment by HolgerReiseVSys** [comment]
Needs to be fixed, too: `org.ehrbase.rest.ehrscape.controller.BaseController::getBaseEnvLinkURL`

File: rest-openehr/src/main/java/org/ehrbase/rest/BaseController.java (hunk 4 above)
Comment (dmitrytulba) [reply rest-openehr/src/main/java/org/ehrbase/rest/BaseController.java:145]: done

File: rest-openehr/src/main/java/org/ehrbase/rest/BaseController.java (hunk 3 above)
Comment (dmitrytulba) [reply rest-openehr/src/main/java/org/ehrbase/rest/BaseController.java:140]: done

File: rest-openehr/src/main/java/org/ehrbase/rest/BaseController.java (hunk 2 above)
Comment (dmitrytulba) [reply rest-openehr/src/main/java/org/ehrbase/rest/BaseController.java:125]: removed

**Review by HolgerReiseVSys — APPROVED** [review]

**Pull request merged**